network/dag/state_test.go

```diff
@@ -340,7 +342,7 @@ func TestState_treeObserver(t *testing.T) {
 	}
 	ctx := context.Background()
 
-	t.Run("no callback for public transaction without payload", func(t *testing.T) {
+	t.Run("callback for public transaction without payload", func(t *testing.T) {
```

Comment:
Why is this allowed? Since v1 public TXs must always have payload?

Reply:
The behaviour changes so the test changed.